src/libstd/sys/windows/fs.rs

```diff
@@ -745,3 +910,78 @@ fn symlink_junction_inner(target: &Path, junction: &Path) -> io::Result<()> {
                                ptr::null_mut())).map(|_| ())
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use prelude::v1::*;
+    use env;
+    use fs::{self, File};
+    use path::PathBuf;
+    use rand::{self, Rng};
+    use sys::c;
+
+    macro_rules! check { ($e:expr) => (
+        match $e {
+            Ok(t) => t,
+            Err(e) => panic!("{} failed with: {}", stringify!($e), e),
+        }
+    ) }
+
+    macro_rules! error { ($e:expr, $s:expr) => (
+        match $e {
+            Ok(_) => panic!("Unexpected success. Should've been: {:?}", $s),
+            Err(ref err) => assert!(err.to_string().contains($s),
+                                    format!("`{}` did not contain `{}`", err, $s))
+        }
+    ) }
+
+    pub struct TempDir(PathBuf);
```

Comment:
Perhaps this could move to a `sys_common::tempdir` module to share between `std::fs::test` and here?